flexmeasures/data/services/utils.py

```diff
@@ -0,0 +1,90 @@
+import functools
+from flask import current_app
+from rq.job import Job
+
+import hashlib
+import base64
+
+from flexmeasures.data.models.time_series import Sensor
+
+
+def make_hash_sha256(o):
+    """
+    SHA256 instead of Python's hash function because apparently, python native hashing function
+    yields different results on restarts.
+    Source: https://stackoverflow.com/a/42151923
+    """
+    hasher = hashlib.sha256()
+    hasher.update(repr(make_hashable(o)).encode())
+    return base64.b64encode(hasher.digest()).decode()
+
+
+def make_hashable(o):
+    """
+    Function to create hashes for dictionaries with nested objects
+    Source: https://stackoverflow.com/a/42151923
+    """
+    if isinstance(o, (tuple, list)):
+        return tuple((make_hashable(e) for e in o))
+
+    if isinstance(o, dict):
+        return tuple(sorted((k, make_hashable(v)) for k, v in o.items()))
+
+    if isinstance(o, (set, frozenset)):
+        return tuple(sorted(make_hashable(e) for e in o))
+
+    if isinstance(o, Sensor):
+        return tuple(
+            (
+                make_hashable(getattr(o, attr, None))
+                for attr in ["attributes", "annotations", "id"]
+            )
+        )
+
+    return o
+
+
+def hash_function_arguments(args, kwags):
+    return make_hash_sha256(args) + make_hash_sha256(kwags)  # concat two hashes
```

Comment:
I've just double checked this. Apparently, the right way is to do h(x,y) = hash(hash(x) || hash(y))  because it avoid the following:

1) h(x,y) = hash(x || y), might create a collision if we delete the last n chracters of x and we append them in front of y. e.g h("abc", "d") = h("ab", "cd")
2) we don't want to sort x and y, because we need the function h(x,y) != h(y,x)
3) extra hashing just avoid that we can't decompose the input arguments and track if the same args or kwarg are called several times. More of a security measure I think.

Reply:
Good to add some of that info to the docstring. Maybe even with a link to your reference.